src/plugins/wip/index.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * @fileoverview Handle PR status for works in progress, using either "WIP" in title or "do not merge" label
+ * @author Kevin Partington
+ */
+
+"use strict";
+
+const WIP_IN_TITLE_REGEX = /^WIP:|\(WIP\)/i;
+const DO_NOT_MERGE_LABEL = "do not merge";
+
+/**
+ * Create status on the PR
+ * @param {Object} context - probot context object
+ * @param {string} state - state can be either success or failure
+ * @param {string} sha - sha for the commit
+ * @param {string} description - description for the status
+ * @param {string} targetUrl The URL that the status should link to
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function createStatusOnPR({ context, state, sha, description, targetUrl }) {
+    return context.github.repos.createStatus(
+        context.repo({
+            sha,
+            state,
+            target_url: targetUrl || "",
+            description,
+            context: "wip"
+        })
+    );
+}
+
+/**
+ * Creates a pending status on the PR to indicate it is a WIP.
+ * @param {Object} context Probot context object
+ * @param {string} sha The SHA hash representing the latest commit to the PR.
+ * @returns {Promise} A Promise that will fulfill when the status check is created
+ */
+async function createPendingWipStatusOnPR(context, sha) {
+    return createStatusOnPR({
+        context,
+        sha,
+        state: "pending",
+        description: "This PR appears to be a work in progress"
+    });
+}
+
+/**
+ * Creates a pending status on the PR to indicate it is WIP.
+ * @param {Object} context Probot context object
+ * @param {string} sha The SHA hash representing the latest commit to the PR.
+ * @returns {Promise} A Promise that will fulfill when the status check is created
+ */
+async function createSuccessWipStatusOnPR(context, sha) {
+    return createStatusOnPR({
+        context,
+        sha,
+        state: "success",
+        description: "This PR is no longer a work in progress"
+    });
+}
+
+/**
+ * Check to see if there is an existing pending wip status check on the PR.
+ * If so, create a success wip status check.
+ * @param {Object} context Probot context object
+ * @param {string} sha Commit SHA hash associated with the status check
+ * @returns {Promise} A Promise which will resolve when a success status check
+ * is created on the PR, or an immediately-resolved Promise if no status check
+ * is needed.
+ */
+async function maybeResolveWipStatusOnPR(context, sha) {
+    const repoAndRef = context.repo({
+        ref: sha
+    });
+
+    let statusCheckExists = false;
+
+    context.github.paginate(
```

Comment:
This needs to be `await` ed otherwise execution will flow through and `statusCheckExists` will always be `false`.
Havent tested it but i am pretty sure that is whats gng to happen. I recommend looking into this.

Reply:
Also why are we paginating?